Add wellness section with links and phone numbers V2

aelnosu opened the pull request:
This PR adds a tab in the sidebar for wellness-related resources.

aelnosu commented:
Thanks for the information Leo. Would it be possible to push the Crisis Support Tab up a level (ie. make it the same level as Courses and Staff so it can be more easily accessed by students)?

aelnosu committed:
Add wellness page as a tab in utilities section.

This is only a placeholder for now. As most resources are not ready yet, I just added a dummy card.

aelnosu commented:
New Version
<img width="3213" height="1897" alt="image" src="https://github.com/user-attachments/assets/044bc672-5719-4094-801a-351dd03d70a4" />

aelnosu commented:
Mobile test
<img width="3213" height="1897" alt="image" src="https://github.com/user-attachments/assets/d04c6d6c-0092-4f23-84b0-2b5998338054" />

aelnosu commented:
Also work on iPhone 5

aelnosu committed:
Move Crisis Support to standalone Wellness tab and not under Resources

Refactor crisis support resources: remove Support component and integrate its content into PhoneNumber component; update references in Resources and Wellness pages.

aelnosu commented:
<img width="3252" height="1955" alt="image" src="https://github.com/user-attachments/assets/f972eb47-19aa-4bee-9cc8-98c382d4e0af" />
<img width="3252" height="1955" alt="image" src="https://github.com/user-attachments/assets/87bcd4fc-e903-436a-850b-4a7c67f77e75" />
<img width="3252" height="1955" alt="image" src="https://github.com/user-attachments/assets/68ae0fa8-7fbc-4ca6-94ac-bbf0b050fd03" />

aelnosu committed:
Change 911 to 988

Update RED phone number to 988, also change reference from Phone1 to Phone

33tm commented:
Looking great!

aelnosu committed:
Add safety Presentation image to Wellness page

Added safety presenation by wellness as an image to the wellness page.

aelnosu committed:
Add mental health resources section and update crisis support information

Formalized all links and phone numbers in support cards

aelnosu commented:
This part of the code is ready for review. I will probably need to add another card, but that will be done when I have time™

aelnosu committed:
Added information about supporting others in crisis.

Added a new section "Support Others" with relevant resources and contact information.

33tm commented on client/src/App.tsx:
```diff
@@ -154,14 +155,13 @@ export default function App() {
                         />
-                        <Route path="/schoology/auth" element={<SgyAuthRedirect />}/>
-                        <Route path="*" element={<PageNotFound />}/>
+                        <Route path="/schoology/auth" element={<SgyAuthRedirect />} />
+                        <Route path="*" element={<PageNotFound />} />
```
```suggestion
                        <Route path="/schoology/auth" element={<SgyAuthRedirect />}/>
                        <Route path="*" element={<PageNotFound />}/>
```

33tm commented on client/src/components/wellness/Presentation.tsx:
From the Gunn website--

https://docs.google.com/presentation/d/e/2PACX-1vQcCWQFnTvmHiH13jphkIOL5oseIURZLOlBDJtMJ8yr1YBIKmaklI-fsTLUB3Jzg8YuUSNbHDaaWlky/pubembed

Would embed this here rather than loading/updating a large image

aelnosu replied:
I chose this so it can be loaded in the app and is scrollable, which we think is ideall

aelnosu replied:
Would use the infamous .webp for speed

33tm commented on client/src/components/wellness/SupportOthers.tsx:
```diff
@@ -0,0 +1,30 @@
+                    <p className="text-sm text-red-600 dark:text-red-400">Request a crisis intervention team</p>
+                </div>
+            </div>
+            If you are concerned for a friend, classmate, or another student and would like Wellness to check in with them, please email <a href="mailto:gunnwellnessteam@pausd.org">gunnwellnessteam@pausd.org</a> with:
```
encapsulate w/ paragraph for proper spacing

```suggestion
            <p>If you are concerned for a friend, classmate, or another student and would like Wellness to check in with them, please email <a href="mailto:gunnwellnessteam@pausd.org">gunnwellnessteam@pausd.org</a> with:</p>
```

33tm commented on client/src/components/wellness/Website.tsx:
```diff
@@ -0,0 +1,47 @@
+                        Email the wellness team for support, questions, or to set up an appointment.
+                    </SupportCard>
+                    <SupportCard name="Headspace" link="https://headspace.org.au/online-and-phone-support/connect-with-us/">
+                        Talk to a councilor with headspace
```
```suggestion
                        Talk to a counselor with headspace
```

aelnosu replied:
oops

33tm commented on client/src/components/wellness/Website.tsx:
```diff
@@ -0,0 +1,47 @@
+                        Talk to a councilor with headspace
+                    </SupportCard>
+                    <SupportCard name="Kids Help Line" link="https://kidshelpline.com.au/">
+                        Contact Kids Help Line to talk to a councilor 24/7
```
```suggestion
                        Contact Kids Help Line to talk to a counselor 24/7
```

33tm commented on client/src/components/wellness/Website.tsx:
These are Australian resources, are they endorsed by Gunn? If not I don't think it's appropriate to include them in WATT.

aelnosu replied:
Yes, they are endorsed by Gunn & wellness

aelnosu replied:
At least that is what Enzo provided me with

aelnosu replied:
Nope, let me fix them

33tm reviewed (commented):
Looks good for the most part, I'd just check the sources and the phrasing describing them. Wouldn't include sources unless it has a direct relation to Gunn.

Additionally, all the anchors should have a target="_blank" to open in a new tab, preventing redirection away from WATT.

aelnosu commented on client/src/App.tsx:
Ok

aelnosu commented:
Again, sorry for the spelling mistake (┬┬﹏┬┬)

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Add `<p>` for wrapping

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Fix Spelling

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Fix spelling

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu commented on client/src/assets/Wellness-Safety.png:
I don't think an iframe would work. Here is the presentation: https://docs.google.com/presentation/d/e/2PACX-1vQcCWQFnTvmHiH13jphkIOL5oseIURZLOlBDJtMJ8yr1YBIKmaklI-fsTLUB3Jzg8YuUSNbHDaaWlky/pub?start=false&loop=false&delayms=3000&slide=id.g267c38f7ec3_0_49

<img width="3152" height="1912" alt="Image" src="https://github.com/user-attachments/assets/7bdf8910-3a95-4f67-94c3-a1da41226dbd" />

aelnosu replied:
<img width="3152" height="1912" alt="Image" src="https://github.com/user-attachments/assets/ed5e8fa5-b5a3-42d3-8d61-93de11f9f351" />

aelnosu committed:
Replace SafetyPresentation image with multiple Wellness presentation slides

It just take too long to load one big image. Split it into multiple smaller images. Also used webp format for better compression.

aelnosu committed:
Remove unendorsed support resources from PhoneNumber component

Removed phone numebrs that are not officially endorsed by Gunn Wellness

aelnosu committed:
Update mental health resources in Website component

Removed unendorsed external resources and added Care Solace as a recommended service.

aelnosu committed:
Fix className formatting for email links in SupportOthers and Website components

Added className="break-all" to email & website links to ensure long addresses break correctly and do not overflow their containers. Also added _blank and rel="noopener noreferrer" to external links for security and usability.

aelnosu committed:
Remove unnecessary blank line in SupportOthers component

IDK why it was there in the first place

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu committed:
Revert white space change.

Co-authored-by: Leo Jeong <80350771+33tm@users.noreply.github.com>

aelnosu commented:
All changes are made and ready for review. If possible, could you merge it next week Wednesday?

aelnosu committed:
Refactor Presentation component to use window dimensions and update ArticleCard for presentation handling

Use iframe for Presentation component and adjust ArticleCard to handle presentation prop, define size as a percentage of window size, not characters

aelnosu committed:
Remove unused Wellness image assets

aelnosu commented:
<img width="3175" height="1977" alt="image" src="https://github.com/user-attachments/assets/eb71be84-fa1e-4325-8dff-37ba666ac535" />

33tm commented:
I think I found the sources you were using---they seem well formatted as is on the Gunn website; I'm thinking again and maybe this should be changed to a link to the Gunn website? I think there's a benefit of having a single source of truth for crisis resources, especially with how frequent and unpredictable Gunn website is. Maybe it's better to rethink this addition? What do you think?
